libraries/AP_HAL/RCOutput.cpp

```diff
@@ -68,3 +68,37 @@ bool AP_HAL::RCOutput::is_dshot_protocol(const enum output_mode mode)
         return false;
     }
 }
+
+/*
+ * calculate the prescaler required to achieve the desire bitrate
+ */
+uint32_t AP_HAL::RCOutput::calculate_bitrate_prescaler(uint32_t timer_clock, uint32_t target_frequency, bool choose_nearest, bool is_dshot)
+{
+    uint32_t prescaler;
+    if (is_dshot) {
+        // original prescaler calculation from betaflight. bi-dir dshot is incredibly sensitive to the bitrate
+        prescaler = uint32_t(lrintf((float) timer_clock / target_frequency + 0.01f) - 1);
+    } else {
+        // adjust frequency to give an allowed value given the clock, erring on the high side
+        prescaler = timer_clock / target_frequency;
+        while ((timer_clock / prescaler) < target_frequency && prescaler > 1) {
+            prescaler--;
+        }
+    }
+
+    // if the closest value is requested then return that
+    if (choose_nearest) {
```

Comment:
is there any case where we don't want choose_nearest?

Reply:
I don't think so. Could just remove the option